Sources/CocoaLumberjack/DDFileLogger.m

```diff
@@ -887,15 +887,16 @@ - (void)lt_rollLogFileNow {
     _currentLogFileHandle = nil;
 
     _currentLogFileInfo.isArchived = YES;
-    NSString *archivedFilePath = [_currentLogFileInfo.filePath copy];
-    _currentLogFileInfo = nil;
 
     if ([_logFileManager respondsToSelector:@selector(didRollAndArchiveLogFile:)]) {
+        NSString *archivedFilePath = [_currentLogFileInfo.filePath copy];
         dispatch_async(_completionQueue, ^{
             [self->_logFileManager didRollAndArchiveLogFile:archivedFilePath];
         });
     }
 
+    _currentLogFileInfo = nil;
```

Comment:
This might become a problem if `didRollAndArchiveLogFile:` accesses `currentLogFileInfo`. This was the reason why we first copied the path, set the `_currentLogFileInfo` to `nil`, and then informed the manager.
You could, however, perform the copy only if the manager responds to the selector, saving any wasted memory due to an unnecessary copy.